src/shared/types.ts

```diff
@@ -25,9 +25,40 @@ export interface User {
   last_name: string;
   email: string;
   is_staff: boolean;
+  is_editor: boolean;
   writer_name: string;
 }
 
+export interface UserAPIError {
+  detail?: string[];
+  user?: string[];
+  username?: string[];
+  email?: string[];
+  password?: string[];
+  writer_name?: string[];
+  [key: string]: string[] | undefined;
+}
+
+export interface UserAPIResponse {
```

Comment:
Like the idea, but we should make it generic and use it API-wide. I.E, `APIResponse<T> { error: <some error obj>, data: T, success: boolean }`

Reply:
57443008

In the future, we probably should change the API to return the user in the data field.

Reply:
DRF let's you customize the final output of all API endpoints. I did the same trick at a co-op once, I'll look into it.